src/java/org/apache/cassandra/db/compaction/AbstractCompactionTask.java

```diff
@@ -66,10 +64,13 @@ protected AbstractCompactionTask(CompactionRealm realm, LifecycleTransaction tra
 
         try
         {
-            // enforce contract that caller should mark sstables compacting
-            Set<SSTableReader> compacting = transaction.getCompacting();
-            for (SSTableReader sstable : transaction.originals())
-                assert compacting.contains(sstable) : sstable.getFilename() + " is not correctly marked compacting";
+            if (!transaction.isOffline())
+            {
+                // enforce contract that caller should mark sstables compacting
+                var compacting = realm.getCompactingSSTables();
+                for (SSTableReader sstable : transaction.originals())
+                    assert compacting.contains(sstable) : sstable.getFilename() + " is not correctly marked compacting";
```

Comment:
this assertion fails on CNDB
I had to disable assertions in order to write unit tests

https://github.com/riptano/cndb/pull/11928/files#diff-5e808a158d1a88c0ead66daeb66d1f6e720b8db9a2e2528079e28a0b3aa0cbd1R94